packages/hash/api/src/model/knowledge/linkEntity.model.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  GraphApi,
+  Entity,
+  LinkEntityMetadata,
+  Filter,
+  EntityStructuralQuery,
+} from "@hashintel/hash-graph-client";
+
+import { EntityModel, EntityTypeModel, LinkEntityModel } from "../index";
+
+export type LinkModelConstructorParams = {
+  linkEntity: Entity;
+  linkEntityTypeModel: EntityTypeModel;
+  leftEntityModel: EntityModel;
+  rightEntityModel: EntityModel;
+};
+
+export type LinkModelCreateParams = {
```

Comment:
Do we want to make these types also match the structure? (So grouping things like the linkMetadata, etc.)

Reply:
So something like this?
```TypeScript
export type LinkModelCreateParams = {
  properties?: EntityProperties;
  metadata: {
    linkEntityTypeModel: EntityTypeModel;
    ownedById: string;
    linkMetadata: {
      leftEntityModel: EntityModel;
      leftOrder?: number;
      rightEntityModel: EntityModel;
      rightOrder?: number;
    };
  };
  actorId: string;
};
```

I think this would be a lot less ergonomic to call, but if you feel strongly I'll address.

Reply:
Would it make it much less ergonomic? In my mind it would make it easier in a lot of places, or more consistent in others. If you were making a new link similar to an old one for example, you could just pass in some of its components like `old.linkMetadata` rather than needing to destructure all of them.

Reply:
I don't feel super strongly, it's just I'd like to see us staying away from the previous state of things where we had umpteen different representations of the exact same object and data. If we have a structure we want for this thing, why not use it in places where we have that data

Reply:
It's just going to add 2 lines everywhere we are creating links to define the `metadata: { linkMetadata: { ... } }` structure. And this wouldn't be the exact same representations of data as the method accepts model classes, rather than identifiers.